### PR TITLE
fix: two phased training fixes

### PR DESCRIPTION
**Pull request by alimaredia**
Phased training is broken, and this commit has the following fixes for it:

1. Phased training does not require a valid data-dir. It has phased training flags and config for the data-dir in phase 1 and phase 2.

2. The eval section from the config was trying to be read from `ctx.obj`, not `ctx.obj.config`. This lead to errors saying the `ctx.obj` does not have a `evaluate` member.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Review by cdoern — APPROVED** [review]
this is fine, but we need to abstract this better in the future so that we aren't calling the map function multiple times

File: src/instructlab/model/train.py [hunk 1]
```diff
@@ -502,6 +500,13 @@ def get_files(directory: str, pattern: str) -> list[str]:
                 "Data for both phase1 and phase2 must be specified for phased training."
             )
 
+        # pull the trainrandom.randinting and torch args from the flags
```
Comment (Maxusmusti) [inline src/instructlab/model/train.py:503]: Typo

**Comment by cdoern** [comment]
once https://github.com/instructlab/instructlab/actions/runs/11072184232/job/30765958698 passes, I will merge this

File: src/instructlab/model/train.py (hunk 1 above)
Comment (Maxusmusti) [reply src/instructlab/model/train.py:503]: guessing that originated from line 484

**Comment by nathan-weinberg** [comment]
> once https://github.com/instructlab/instructlab/actions/runs/11072184232/job/30765958698 passes, I will merge this

There's an outstanding comment from @Maxusmusti and @JamesKunstle should really look at this

**Comment by cdoern** [comment]
e2e passed, but we might instead go with @JamesKunstle 's #2236 since it fixes the same issues

**Comment by nathan-weinberg** [comment]
Superceded by https://github.com/instructlab/instructlab/pull/2236

**Pull request closed without merging**